streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -879,7 +879,10 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
      * A {@link Transformer} (provided by the given {@link TransformerSupplier}) is applied to each input record and
      * returns zero or one output record.
      * Thus, an input record {@code <K,V>} can be transformed into an output record {@code <K':V'>}.
-     * This is a stateful record-by-record operation (cf. {@link #map(KeyValueMapper) map()}).
+     * This is a stateful record-by-record operation (cf. {@link #map(KeyValueMapper) map()}). Note that it is not
+     * automatically stateful, and will be stateless if you choose not to attach a state store. In this case it is
+     * similar to {@link #map(KeyValueMapper) map()} but provides access to the {@code ProcessorContext} and
+     * associated metadata.
```

Comment:
It might be worth mentioning that this is also a way of mixing in PAPI functionality to the DSL, but I'll leave that up to you.  This comment applies to here and below.

Reply:
Yeah I did a search for that and saw it's mentioned in the `KStream` javadocs, but not here. I think I agree that it's worth reiterating in the transformer docs however